src/rgw/rgw_rados.h

```diff
@@ -3600,6 +3600,10 @@ class RGWRados
       (get_zonegroup().zones.size() > 1 || current_period.is_multi_zonegroups_with_zones());
   }
 
+  bool can_reshard() {
+    return get_zonegroup().zones.size() == 1;
```

Comment:
this could be a secondary zonegroup with a single zone, which would still be replicating the bucket instance from the meta master. so i think we also want to check that there's only one zonegroup in the period, ie `RGWPeriod::is_single_zonegroup()`